test/AutoRest.Shared.Tests/RawRequestUriBuilderTest.cs

```diff
@@ -12,7 +12,30 @@ public class RawRequestUriBuilderTest
         [TestCase("http://|local|host", "override", null, "http://override/")]
+        [TestCase("http://|local|host|:12345", null, null, "http://localhost:12345/")]
+        [TestCase("http://|local|host:12345", null, null, "http://localhost:12345/")]
         [TestCase("http://localhost/with|some|path", null, "/andMore", "http://localhost/withsomepath/andMore")]
+        [TestCase("http://localhost:12345/with|some|path", null, "/andMore", "http://localhost:12345/withsomepath/andMore")]
+        [TestCase("http://localhost|:12345/with|some|path", null, "/andMore", "http://localhost:12345/withsomepath/andMore")]
+        [TestCase("http://|localhost:12345|/with|some|path", null, "/andMore", "http://localhost:12345/withsomepath/andMore")]
+        [TestCase("http://|localhost:12345|withsomepath", null, "/andMore/", "http://localhost:12345/withsomepath/andMore/")]
+        [TestCase("http://|localhost:12345/|withsomepath", null, "/andMore/", "http://localhost:12345/withsomepath/andMore/")]
+        [TestCase("http://localhost|/more|/andMore|/evenMore", null, null, "http://localhost/more/andMore/evenMore")]
+        [TestCase("http://localhost|/more/andMore|/evenMore/", null, null, "http://localhost/more/andMore/evenMore/")]
+        [TestCase("http://|localhost:12345|/more/andMore|/evenMore/", null, null, "http://localhost:12345/more/andMore/evenMore/")]
+        [TestCase("http://localhost|/more|/andMore|?one=1", null, null, "http://localhost/more/andMore?one=1")]
+        [TestCase("http://localhost|/more|/andMore?one=1", null, null, "http://localhost/more/andMore?one=1")]
+        [TestCase("http://localhost/|more?one=1", null, null, "http://localhost/more?one=1")]
+        [TestCase("http://localhost|/more?one=1", null, "/andMore", "http://localhost/more/andMore?one=1")]
+        [TestCase("http://localhost|/more?one=1|&two=2|&three=3", null, null, "http://localhost/more?one=1&two=2&three=3")]
+        [TestCase("http://localhost/more?one=1&two=2|&three=3", null, null, "http://localhost/more?one=1&two=2&three=3")]
+        [TestCase("http://localhost/more|?one=1&two=2|&three=3", null, null, "http://localhost/more?one=1&two=2&three=3")]
+        [TestCase("http://localhost/more|?one=1|&two=2&three=3", null, null, "http://localhost/more?one=1&two=2&three=3")]
+        [TestCase("http://localhost|/more?one=1&two=2&three=3", null, null, "http://localhost/more?one=1&two=2&three=3")]
+        [TestCase("http://|localhost:12345|/more|?one=1|&two=2|&three=3", null, null, "http://localhost:12345/more?one=1&two=2&three=3")]
+        [TestCase("http://|localhost|:12345|/more|?one=1|&two=2|&three=3", null, null, "http://localhost:12345/more?one=1&two=2&three=3")]
+        [TestCase("http://localhost|/more?one=&two=&three=", null, null, "http://localhost/more?one=&two=&three=")]
+        [TestCase("http://localhost|/more?one&two&three", null, null, "http://localhost/more?one=&two=&three=")]
```

Comment:
Do we support `[TestCase("http://localhost|/more?one&&&&&&&three", null, null, "http://localhost/more?one=&two=&three=")]` ?

Reply:
I've added a test for it now. It works as expected. 😁